### PR TITLE
simplify pom, remove non used dependencies and use bom for LTS we use. need some changes in mockito

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

### CODE REVIEW

**Comment by olamy** [comment]
still in draft as using mockito core version from jenkins bom need some changes. But this should fix PCT failures with jdk11

**Comment by warden** [comment]
I actually did similar changes locally in my feature PR but got stuck in on the Mockito update with the unexpected number of issues to be fixed with it,  so that would definitely help!

File: pom.xml [hunk 1]
```diff
@@ -119,6 +111,18 @@
...
        </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <!-- not reuse because Mockito record stuff statically and some are changing between tests so it suc... -->
+          <reuseForks>false</reuseForks>
+          <!-- to run some tests within your IDE you will need this -->
+<!--          <systemPropertyVariables>-->
```
Comment (warden) [inline pom.xml:121]: Oh, now it's clear - how about adding this to a profile?

File: src/main/java/hudson/plugins/jira/JiraJobAction.java [hunk 2]
```diff
@@ -31,7 +32,7 @@
  * e.g. "JENKINS-1234" and "feature/JENKINS-1234" will have this action with the issue JENKINS-1234 referenced
  */
 @ExportedBean
-public class JiraJobAction implements Action {
+public class JiraJobAction implements Action, Serializable {
```
Comment (warden) [inline src/main/java/hudson/plugins/jira/JiraJobAction.java:35]: Why is this required

Comment (olamy) [reply src/main/java/hudson/plugins/jira/JiraJobAction.java]: good catch sorry I miss to remove this. done

Comment (warden) [reply src/main/java/hudson/plugins/jira/JiraJobAction.java]: Please organise imports - https://sonarcloud.io/project/issues?id=jenkins-jira-plugin&open=AXIIy9Pg99-5isQx0vpU&pullRequest=239&resolved=false&types=CODE_SMELL

**Comment by warden** [comment]
LGTM, ready for squash merge?
